src/PageStreamingDescriptor.php

```diff
@@ -55,6 +57,13 @@ public function __construct($descriptor)
     {
         self::validate($descriptor);
         $this->requestPageTokenField = $descriptor['requestPageTokenField'];
+        if (isset($descriptor['requestPageSizeField'])) {
+            $this->requestHasPageSizeField = true;
+            $this->requestPageSizeField = $descriptor['requestPageSizeField'];
+        } else {
+            $this->requestHasPageSizeField = false;
+            $this->requestPageSizeField = null;
```

Comment:
Couldn't you just get by with only requestPageSizeField, and distinguish its presence/absence using null vs not null?